### PR TITLE
Feature previews

### PR DESCRIPTION
**Pull request by andrzejewsky**
AC:
- All available feature flags (feature previews) should be available within the modal, toggled from the `user menu -> feature previews`
- User should be able to enable or disable the feature
- Developer is able to create markdown file with new feature along with its description and image
- Developer can enable flag by default by setting `enable: true` within markdown attributes section
- When developer set `visible: false` in markdown, flag will not be visible in the feature preview modal
- When there is no flag, modal shows proper information about it

**Notes**:
_after making changes within markdown files (add new one, edit etc.) please run `npm run build-flags` - it will generate the content_

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core
6. [ ] I added changesets and [read good practices](../.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] app
3. [ ] attribute
4. [ ] category
5. [ ] collection
6. [ ] customer
7. [ ] giftCard
8. [ ] homePage
9. [ ] login
10. [ ] menuNavigation
11. [ ] navigation
12. [ ] orders
13. [ ] pages
14. [ ] payments
15. [ ] permissions
16. [ ] plugins
17. [ ] productType
18. [ ] products
19. [ ] sales
20. [ ] shipping
21. [ ] translations
22. [ ] variants
23. [ ] vouchers

CONTAINERS=2

### CODE REVIEW

**Review by krzysztofzuraw — COMMENTED** [review]
Shouldn't provider be added to storybook as well?

**Comment by andrzejewsky** [comment]
@krzysztofzuraw https://github.com/saleor/saleor-dashboard/blob/main/.storybook/decorators/MockedProvidersDecorator.tsx#L38 it already is

**Review by krzysztofzuraw — DISMISSED** [review]

File: src/components/Sidebar/user/FeatureFlagsModal/usePersistance.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import {
+  flagInfoToFlagList,
+  Name,
+} from "@dashboard/featureFlags/availableFlags";
+import { flagListToMetadata } from "@dashboard/featureFlags/strategies/MetadataStrategy";
+import { useFlagsInfo } from "@dashboard/featureFlags/useFlagsInfo";
+import { useUserAccountUpdateMutation } from "@dashboard/graphql";
+
+export const usePersistance = () => {
```
Comment (Droniu) [inline src/components/Sidebar/user/FeatureFlagsModal/usePersistance.ts:9]: typo (also change filename)
```suggestion
export const usePersistence = () => {
```

**Review by Droniu — APPROVED** [review]

**Comment by wojteknowacki** [comment]
The feature preview button should always open a list of feature flags - now user need to release user account modal to get back to list view
https://github.com/saleor/saleor-dashboard/assets/124166231/2be62d8f-4b97-46d7-8c47-373aee5df4b2

**Review by krzysztofzuraw — DISMISSED** [review]

**Comment by wojteknowacki** [comment]
- When multiple flags are present list is not scrollable
- When log text added as description last line is fully visible
image

**Review by wojteknowacki — APPROVED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Pull request merged**